pandas/core/groupby.py

```diff
@@ -534,19 +534,18 @@ def curried(x):
             except Exception:
                 try:
                     return self.apply(curried)
-                except Exception:
+                except Exception as e:
 
                     # related to : GH3688
                     # try item-by-item
-                    # this can be called recursively, so need to raise
-                    # ValueError
-                    # if we don't have this method to indicated to aggregate to
-                    # mark this column as an error
-                    try:
-                        return self._aggregate_item_by_item(name,
-                                                            *args, **kwargs)
-                    except (AttributeError):
-                        raise ValueError
+                    # this can be called recursively, so need to raise a
```

Comment:
If you don't like this proposition, I'll do that... but I wanted to try this last thing first, because I'd feel better having made the general code somewhat better rather than patching with an ad-hoc solution for `.rank`.

Because, right now, I feel the code is flawed. Maybe it's just that I am not experienced enough to understand the whole architecture, but... isn't it a bit hazardous to use generic exceptions such as `ValueError` to signal stuff internally, and then catch all the `ValueError` blindly? Here the result (if I'm not wrong) is that we try to rank on each item, but each fails, but we catch all the exceptions blindly — because we are not able to distinguish between the things that fail because they should fail and need therefore to be discarded, and the things that fail because there is a problem that need to be signalled to the user — and so we end up with an empty dataframe.

So what I did here was just to make sure that at least we would not forget any meaningful error message (hence the `ValueError(e)` below), and then when we end up with an empty dataframe, reraise the last error. This is the most simple way I could find, so if you find it weird I'm not going to keep pushing, it just means I still have to learn a lot :)

What is it that you don't like? 

BTW, I understand I'm taking your time by trying stuff like that and expecting you to have a look each time, I appreciate. So feel free to tell me if you think my approach is taking too much energy for both of us, I'll do a patch for `.rank`, and maybe start another issue for a more comprehensive solution. But maybe, if I can understand better what is it that seems wrong to you, and if it is a minor thing, I can still fix it? ^^